clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -487,7 +487,12 @@ else if (strategy == OffsetResetStrategy.LATEST)
                 } else {
                     TopicPartition partition = nextInLineRecords.partition;
                     fetchedPartition = partition;
-                    fetchedOffsets = subscriptions.position(partition);
+                    if (subscriptions.isAssigned(partition)) {
```

Comment:
I took a closer look and it seems like this is a regression from https://github.com/apache/kafka/commit/a26731674815a57368ad4500a77e84a218a40e63#diff-b45245913eaae46aa847d2615d62cde0R490 so we may need another RC.

Maybe we need to move the `position` call to inside `!records.isEmpty` since `drainRecords` does the check. But not sure, better to wait for @hachikuji to take a look.

cc @becketqin @lindong28 @hachikuji